app/jobs/export_job.rb

```diff
@@ -23,16 +26,14 @@ def perform(config)
 
     # Construct export
     lookups = []
+
     patients = patients_by_query(user, data.dig(:patients, :query) || {})
 
-    # Custom export is already sorted by id, calling order on custom export patients leads to invalid SQL statement because id is not in select list
-    patients = patients.order(:id) unless config[:export_type] == :custom
-    patients.find_in_batches(batch_size: RECORD_BATCH_SIZE).with_index do |patients_group, index|
-      # Duplicate the config as it gets changed in the following method calls and should be fresh each batch.
-      config_dup = config.deep_dup
-      exported_data = get_export_data(patients_group, config_dup[:data])
-      files = write_export_data_to_files(config_dup, exported_data, index)
-      lookups.concat(create_lookups(config_dup, files))
+    # NOTE: The reorder here clears out any other sorting that may have been added to this query as it should just be sorting by ID when
+    # getting batches. in_batches appears to NOT sort within batches, so ordering is also done deeper down.
+    patients.reorder('').in_batches(of: OUTER_BATCH_SIZE).each_with_index do |patients_group, index|
```

Comment:
If patients does not return anything, such as the case of export purge eligible, the user will be confused because the export starts and finishes successfully but no email is sent letting the user know that nothing was actually available to export.

Seems like we would need to handle the case where patients is empty and send an email saying nobody matched the export criteria.

Reply:
Agreed, but out of scope of this PR since that's been the functionality. Made a ticket for this here: https://tracker.codev.mitre.org/browse/SARAALERT-1186